workflow/snakemake_rules/main_workflow.smk

```diff
@@ -96,60 +96,72 @@ rule diagnose_excluded:
             --output-exclusion-list {output.to_exclude} 2>&1 | tee {log}
         """
 
-rule prefilter:
-    message:
-        """
-        Pre-filtering sequences before aligning for minimal length.
-         - Input: {input.sequences}
-         - Filtering to sequences with more than {params.min_length}bp.
-         - Output: {output.sequences}
-        """
-    input:
-        sequences = lambda wildcards: _get_path_for_input("sequences", wildcards.origin),
-        metadata = lambda wildcards: _get_path_for_input("metadata", wildcards.origin)
-    output:
-        sequences = "results/prefiltered{origin}.fasta"
-    log:
-        "logs/prefiltered{origin}.txt"
-    params:
-        min_length = lambda wildcards: _get_filter_value(wildcards, "min_length")
-    conda: config["conda_environment"]
-    shell:
-        """
-        augur filter \
-            --sequences {input.sequences} \
-            --metadata {input.metadata} \
-            --min-length {params.min_length} \
-            --output {output.sequences} 2>&1 | tee {log}
-        """
-
-rule align:
-    message:
-        """
-        Aligning sequences from {input.sequences} to {input.reference}
-          - gaps relative to reference are considered real
-        """
-    input:
-        sequences = lambda wildcards: _get_path_for_input("prefiltered", wildcards.origin),
-        reference = config["files"]["alignment_reference"]
-    output:
-        alignment = "results/aligned{origin}.fasta"
-    log:
-        "logs/align{origin}.txt"
-    benchmark:
-        "benchmarks/align{origin}.txt"
-    threads: 16
-    conda: config["conda_environment"]
-    shell:
-        """
-        mafft \
-            --auto \
-            --thread {threads} \
-            --keeplength \
-            --addfragments \
-            {input.sequences} \
-            {input.reference} > {output} 2> {log}
-        """
+if "use_nextalign" in config and config["use_nextalign"]:
+    rule align:
+        message:
+            """
+            Aligning sequences to {input.reference}
+              - gaps relative to reference are considered real
+            """
+        input:
+            sequences = lambda wildcards: _get_path_for_input("sequences", wildcards.origin),
+            genemap = config["files"]["annotation"],
+            reference = config["files"]["alignment_reference"]
+        output:
+            alignment = "results/aligned{origin}.fasta",
+            insertions = "results/insertions{origin}.tsv",
+            translations = expand("results/translations/seqs{{origin}}.gene.{gene}.fasta", gene=config.get('genes', ['S']))
+        params:
+            outdir = "results/translations",
+            bin = config["nextalign_bin"],
+            genes = ','.join(config.get('genes', ['S'])),
+            basename = "seqs{origin}"
+        log:
+            "logs/align{origin}.txt"
+        benchmark:
+            "benchmarks/align{origin}.txt"
+        threads: 8
```

Comment:
there is a modest benefit of using 16cores. I reduced it to 8 because scheduling became easier. I think CPU utilization with 8 cores is 650% and more like 1000% with 16 cores.